src/Stratis.Bitcoin/Base/Deployments/ThresholdConditionCache.cs

```diff
@@ -111,8 +114,15 @@ public List<ThresholdStateModel> GetThresholdStateMetrics(ChainedHeader indexPre
                     indexPrev = indexPrev.Previous;
                 }
 
+                // look in the cache for the hash of the first block an item was deployed
+
+                var firstSeenHash = this.cache.Reverse().FirstOrDefault(c => c.Value[deploymentIndex] != null);
+
+                int sinceHeight = indexPrev.FindAncestorOrSelf(firstSeenHash.Key).Height;
```

Comment:
Possible NullReferenceException for `firstSeenHash`

Reply:
hmm in reality no there should always be a value I will remove the `Default` part

Reply:
actually fix as suggested